internal/tiltfile/tiltfile_state.go

```diff
@@ -1331,28 +1331,44 @@ type k8sObjectSelector struct {
 // Creates a new k8sObjectSelector
 // If an arg is an empty string it will become an empty regex that matches all input
 // Otherwise the arg must match the input exactly
-func newExactK8sObjectSelector(apiVersion string, kind string, name string, namespace string) (k8sObjectSelector, error) {
-	ret, err := newK8sObjectSelector(
-		exactOrEmptyRegex(apiVersion),
-		exactOrEmptyRegex(kind),
-		exactOrEmptyRegex(name),
-		exactOrEmptyRegex(namespace),
-	)
+func newExactCaseInsensitiveK8sObjectSelector(apiVersion string, kind string, name string, namespace string) (k8sObjectSelector, error) {
```

Comment:
I feel like the nomenclature here needs to be cleaned up a bit, in particular:
- All functions are case insensitive now, but only one of them has "CaseInsensitive" in the name. That seems weird?
- The usage of the word "exact" here is not something i've seen before. I think python calls this "FullMatch"?

Reply:
ooh I like FullMatch. And I agree, I'll clean this stuff up.